initialization/initialization.go

```diff
@@ -167,13 +162,8 @@ func (init *Initializer) Stop() error {
 }
 
 func (init *Initializer) SessionNumLabelsWrittenChan() <-chan uint64 {
-	init.mtx.Lock()
-	defer init.mtx.Unlock()
-
-	if init.numLabelsWrittenChan == nil {
-		init.numLabelsWrittenChan = make(chan uint64, 1024)
-	}
-
+	init.mtx.RLock()
+	defer init.mtx.RUnlock()
 	return init.numLabelsWrittenChan
```

Comment:
Why is a lock required here?

Reply:
Creation of the channel is secured by a lock, I wanted to avoid someone getting `nil` here.

Reply:
It leads to the conclusion that if `SessionNumLabelsWrittenChan()` is called before the channel is created in `Initialize()`, the caller will receive `nil` or a closed channel (from a previous run). I looked up how this is used in `PostSetupManager` in go-spacemesh`. It is exposed via the `PostSetupProvider` interface and then used from the GRPC API. It will break if the GRPC API user call `PostSetupStatusStream()` before calling `StartSmeshing`.

IMO, `Initializer` (and `PostSetupManager`) could be made much simpler and robust if it was not made "reusable". 

`PostSetupManager::StartSession()` already creates a completely new instance of `Initializer` rather than reusing the old one. 

`Initializer::Stop` could be changed to just cancelling a context.

I'm not sure what `Initializer::Reset()` is for.

Reply:
It looks like `Reset()` is used primarily in tests to clean up between tests and by the PostSetupManager to delete the local PoST files if necessary. This is currently only the case in tests as well.

The only use case I see for calling `Reset()` on a live node, is if the node changes identities. In this case it might be better to just have multiple folders, one for each identity, so that changing identities doesn't require re-initialising the PoST for them.

I agree with the point of `Stop()` being unnecessary if we would just pass in a `context` to `Initialize()` and cancel that if we want to stop the Initializer from what it is doing.

Regarding `SessionNumLabelsWrittenChan()`: imho this function isn't needed at all. `PostSetupManager::StatusChan()` should just call `SessionNumLabelsWritten()` and `Started()` `Completed()`, `isInitializing()` or better a single method that combines the purpose of all 3. `PostSetupmanager::StatusChan()` could then just return the current status instead of a channel, which in turn is called by `SmesherService::PostSetupStatusStream()` in regular intervals.

This way `PostSetupStatusStream()` controls how often status updates are sent via the stream instead of `Initializer`. All this methods should return quickly and not interfere with initialization via blocking of channels or similar.

I'll create a ticket for the refactoring, since this should be done separately from this PR.

Reply:
LGTM